src/core/tracing/Window.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Override some bits of foam.core.Window to support more accurate tracing.
+foam.CLASS({
+  package: 'foam.core.tracing',
+  name: 'Window',
+  extends: 'foam.core.Window',
```

Comment:
@kgrgreer, some context: This PR initially included an axiom called "overwrites" that extends a base class (including SUPER support), then registers the new class under the old class's name. After discussing the complexity of this solution, Adam and I agreed that "overwrites" was yet-another-way-to-do-inheritance that we don't want muddying the waters.

In the case of FObject, we switched to "refines" and used constants to store base method definitions (pseudo-SUPER-support). In the case of Window, Adam pointed out that overwriting the global context would probably be enough to capture calls to |merged| and |framed|; we can use SUPER as usual (and we do).

You correctly point out that this means when creating a new window, I have to know that tracing is installed and decide whether or not to use it (create a foam.core.Window or foam.core.tracing.Window). Should we make window tracing a system-wide all-or-nothing kind of thing by implementing the refines+pseudo-SUPER strategy used for FObject? @adamvy, what do you think?